.github/workflows/validate-helm-chart.yml

```
@@ -50,3 +53,31 @@ jobs:
             echo ""
             echo "Helm values documentation is correct!"
           fi
+
+  helm-generation-check:
+    runs-on: ubuntu-latest
+    
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      
+      - name: Generate current YAML
+        run: |
+          helm template helm/chart > current.yaml
+      
+      - name: Generate new Helm Chart
+        run: make helm-package
+
+      - name: Generate new YAML
+        run: |
+          helm template helm/chart > new.yaml
+          
+      - name: Compare YAML files
+        run: |
+          if ! diff -rq current.yaml new.yaml &>/dev/null; then 
+            echo "Helm charts were not re-generated. Please regenerate them using make helm-package"
+            exit 
```

we need a non zero exit code here to make the pipeline fail
```suggestion
            exit 1
```